### PR TITLE
Multiple APIs for faking images

### PR DESCRIPTION
**Pull request by spacedog4**
lorempixel is out of air so I made some change in Faker Image class so I can pass a param and change the base url to loremflickr that works perfectly

### CODE REVIEW

**Comment by loic425** [comment]
This pull request solves that issue #1543

**Comment by loic425** [comment]
@spacedog4 All faker image api don't have same path system.
https://picsum.photos/

With this one, you need to add random query parameter.
imho, you should restrict base url to a compatible list.

**Comment by loic425** [comment]
@spacedog4 another feature could be to automatically switch api when one is down.

**Comment by spacedog4** [comment]
@loic425 I've just pushed an update, now you pass the name of the api you want to use, restricted by four, loremixel, loremflickr, picsumphotos and placeholder, each one have a extends a contract to generate a the URL, it means if anyone wants to add a new api just need to create anothe class that extends this contract and create the imageUrl function. If one api is unsuccessful it automatically tries to use the next one #

**Comment by loic425** [comment]
@spacedog4 ok that's far better :)  Now, waiting for a core team member to review that. I can request you some changes but a core team mmeber review seems to be more appropriate.

**Comment by loic425** [comment]
@spacedog4 But you can already fix some codestyle errors looking on travis logs.

File: src/Faker/Provider/Image.php [hunk 1]
```diff
@@ -7,9 +7,14 @@
  */
 class Image extends Base
 {
-    protected static $categories = array(
-        'abstract', 'animals', 'business', 'cats', 'city', 'food', 'nightlife',
-        'fashion', 'people', 'nature', 'sports', 'technics', 'transport'
+
+    protected static $apis = array(
+        'lorempixel'   => ImageContracts\LoremPixel::class,
```
Comment (loic425) [inline src/Faker/Provider/Image.php]: It's not php5 compatible. Php5 builds fail because of these following lines.

**Comment by fzaninotto** [comment]
I've already said that I won't accept such PRs. I want to remove the image generator, not to add more. See #957 for the rationale.

**Pull request closed without merging**